solutions/usaco-1159.mdx

```diff
@@ -0,0 +1,140 @@
+---
+id: usaco-1159
+source: USACO Silver 2021 December
+title: Connecting Two Barns
+author: Nathan Gong
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob2_gold_open21.html)
+
+## Implementation - DFS + Two Pointers
+
+**Time Complexity:** $\mathcal{O}(T\cdot (N + M))$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class ConnectingTwoBarns {
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int t = io.nextInt();
+		for (int test = 0; test < t; test++) {
+			int n = io.nextInt();
+			int m = io.nextInt();
+
+			List<List<Integer>> adj = new ArrayList<>();
+			for (int i = 0; i < n; i++) adj.add(new ArrayList<>());
```

Comment:
braces

Reply:
also can't `adj` just be an array of arraylists?

Reply:
it can, but i just prefer this so the syntax doesn't change

Reply:
still, braces?